internal/controller/imagerepository_controller.go

```diff
@@ -357,7 +376,12 @@ func (r *ImageRepositoryReconciler) setAuthOptions(ctx context.Context, obj *ima
 		default:
 			opts = r.DeprecatedLoginOpts
 		}
-		auth, authErr = login.NewManager().Login(ctx, obj.Spec.Image, ref, opts)
+		var managerOpts []login.Option
+		if proxyURL != nil {
+			managerOpts = append(managerOpts, login.WithProxyURL(proxyURL))
+		}
```

Comment:
Why not `login.NewManager(login.WithProxyURL(proxyURL)).Login(...)`, the nil check is done by the manager. This is how I did it in source-controller here: https://github.com/fluxcd/source-controller/pull/1607

Reply:
I think `login.NewManager(login.WithProxyURL(proxyURL)).Login(...)` kinda suggests to the reader that there is always a proxy URL set in this case (and is also a long expression which would read nicer if broken down, but that's unrelated to your specific suggestion, we could just break it in `managerOpts := ...; manager := ...; manager.Login(...)` and still do it as you say). This is how I did it for most of these optional parameters for Bucket: https://github.com/fluxcd/source-controller/blob/main/internal/controller/bucket_controller.go#L449-L537